x-pack/plugins/alerting/public/pages/maintenance_windows/components/recurring_schedule_form/custom_recurring_schedule.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import moment from 'moment';
+import { css } from '@emotion/react';
+import { getUseField, useFormData } from '@kbn/es-ui-shared-plugin/static/forms/hook_form_lib';
+import { Field } from '@kbn/es-ui-shared-plugin/static/forms/components';
+import { EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
+import { CREATE_FORM_CUSTOM_FREQUENCY, Frequency, WEEKDAY_OPTIONS } from '../../constants';
+import * as i18n from '../../translations';
+import { ButtonGroupField } from '../fields/button_group_field';
+import { getInitialByWeekday } from '../../helpers/get_initial_by_weekday';
+import { getWeekdayInfo } from '../../helpers/get_weekday_info';
+
+const UseField = getUseField({ component: Field });
+
+const styles = {
+  flexField: css`
+    .euiFormRow__labelWrapper {
+      margin-bottom: unset;
+    }
+  `,
+};
+
+export const CustomRecurringSchedule: React.FC = React.memo(() => {
+  const [{ startDate, recurringSchedule }] = useFormData({
+    watch: [
+      'startDate',
+      'recurringSchedule.frequency',
+      'recurringSchedule.interval',
+      'recurringSchedule.customFrequency',
+    ],
+  });
+
+  const frequencyOptions = useMemo(
+    () => CREATE_FORM_CUSTOM_FREQUENCY(recurringSchedule?.interval),
+    [recurringSchedule?.interval]
+  );
+
+  const bymonthOptions = useMemo(() => {
+    if (!startDate) return [];
+    const date = moment(startDate);
+    const { dayOfWeek, nthWeekdayOfMonth, isLastOfMonth } = getWeekdayInfo(date, 'ddd');
+    return [
+      {
+        id: 'day',
+        label: i18n.CREATE_FORM_CUSTOM_REPEAT_MONTHLY_ON_DAY(date),
+      },
+      {
+        id: 'weekday',
+        label: i18n.CREATE_FORM_WEEKDAY_SHORT(dayOfWeek!)[isLastOfMonth ? 0 : nthWeekdayOfMonth!],
```

Comment:
nitpick: i believe these non-null operators are not needed

Reply:
Resolved in this commit e408455440054bd94d83fc8142a0b01a77860a5c